common.tf

```diff
@@ -0,0 +1,71 @@
+locals {
+  az_count = local.environment == "prod" ? 2 : 1
+}
+resource "random_password" "preservica_password" {
+  length = 20
+}
+
+resource "random_string" "preservica_user" {
+  length  = 10
+  special = false
+}
+
+resource "aws_secretsmanager_secret" "preservica_secret" {
+  name = "${local.environment}-preservica-api-login-details-${random_string.preservica_user.result}"
+  policy = templatefile("./templates/secrets_manager/secrets_manager_lambda_policy.json.tpl", {
+    account_number = var.dp_account_number
+  })
+}
+
+resource "aws_secretsmanager_secret_version" "preservica_secret_version" {
+  secret_id     = aws_secretsmanager_secret.preservica_secret.id
+  secret_string = jsonencode({ (random_string.preservica_user.result) = random_password.preservica_password.result })
+  lifecycle {
+    ignore_changes = [secret_string]
+  }
+}
+
+module "vpc" {
+  source                       = "git::https://github.com/nationalarchives/da-terraform-modules//vpc"
+  vpc_name                     = "${local.environment}-vpc"
+  az_count                     = local.az_count
+  elastic_ip_allocation_ids    = [aws_eip.eip.allocation_id]
+  nat_instance_security_groups = [module.nat_instance_security_group.security_group_id]
+}
+
+resource "aws_eip" "eip" {
+  domain = "vpc"
+}
```

Comment:
DOes 1 EIP allow us to be HA?

Reply:
Good point, I've put a count in with the az_count.